File: common/src/main/java/net/superricky/tpaplusplus/util/Request.java
package net.superricky.tpaplusplus.util;

import net.minecraft.server.level.ServerPlayer;

public class Request {
    private final ServerPlayer sender;
    private final ServerPlayer receiver;
    private final boolean hereRequest;
    private boolean accepted;
    private double acceptX, acceptY, acceptZ;

    public Request(ServerPlayer sender, ServerPlayer receiver, boolean hereRequest) {
        this.sender = sender;
        this.receiver = receiver;
        this.hereRequest = hereRequest;
        this.accepted = false;
    }

    public ServerPlayer getSender() {
        return sender;
    }

    public ServerPlayer getReceiver() {
        return receiver;
    }

    public boolean isHereRequest() {
        return hereRequest;
    }

    public boolean isAccepted() {
        return accepted;
    }

    public double getAcceptX() {
        return acceptX;
    }

    public double getAcceptY() {
        return acceptY;
    }

    public double getAcceptZ() {
        return acceptZ;
    }

    public void setAccepted(boolean accepted) {
        this.accepted = accepted;
    }

    public void setAcceptPosition(double acceptX, double acceptY, double acceptZ) {
        this.acceptX = acceptX;
        this.acceptY = acceptY;
        this.acceptZ = acceptZ;
    }

    @Override
    public String toString() {
        return "Sender:" + sender.getName() + ", Receiver:" + receiver.getName() + ", isHereRequest:" + hereRequest + ", isAccepted:" + accepted;
    }
}

File: src/main/kotlin/net/superricky/tpaplusplus/async/request/Request.kt
package net.superricky.tpaplusplus.async.request

import net.minecraft.server.network.ServerPlayerEntity
import net.superricky.tpaplusplus.async.AsyncCommandType

class Request(
    val sender: ServerPlayerEntity,
    val receiver: ServerPlayerEntity?,
    val commandType: AsyncCommandType,
    val hereRequest: Boolean = false
) {
    var delay: Double = commandType.handler.getDelayTime()
    var cooldown: Double = commandType.handler.getCooldownTime()

    override fun toString(): String =
        "Request{sender=${sender.name}, receiver=${receiver?.name}, commandType=$commandType, hereRequest=$hereRequest}"
}
